src/compile/compile.ts

```diff
@@ -96,15 +96,7 @@ export function assembleNestedMainGroup(model: Model) {
   const {layout, signals, ...group} =  model.assembleGroup([]);
   const marks = group.marks;
 
-  const hasLayout = !!model.assembleLayout();
-  const parentEncodeEntry = {
-    ...(!hasLayout ? {
-      width: {signal: 'width'},
-      height: {signal: 'height'},
-    } : {}),
-    ...model.assembleParentGroupProperties()
-  };
-
+  const parentEncodeEntry = model.assembleParentGroupProperties();
```

Comment:
I think this is can be incorrect, if the root model has `name`, you won't correctly output `width` and `height` signals for this nested main group.